datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -426,52 +426,78 @@ impl Iterator for SortedIterator {
         // Combine adjacent indexes from the same batch to make a slice,
         // for more efficient `extend` later.
         let mut last_batch_idx = 0;
-        let mut start_row_idx = 0;
-        let mut len = 0;
+        let mut indices_in_batch = vec![];
```

Comment:
I don't understand why using a `Vec` here is better. Aren't the indices a contiguous range?

I wonder what about using  `Option<std::ops::Range>` instead https://doc.rust-lang.org/std/ops/struct.Range.html ?

Maybe as a follow on PR